components/molecule/photoUploader/src/photoTools.js

```diff
@@ -0,0 +1,317 @@
+      } else if (inputImageAspectRatio < outputImageAspectRatio) {
+        outputHeight = outputWidth / outputImageAspectRatio
+      }
+
+      canvas.width = outputWidth
+      canvas.height = outputHeight
+
+      switch (rotation) {
+        case 0:
+          context.translate(
+            (outputWidth - inputWidth) * 0.5,
+            (outputHeight - inputHeight) * 0.5
+          )
+          break
+        case 90:
+          context.rotate(0.5 * Math.PI)
+          context.translate(
+            (outputHeight - inputWidth) * 0.5,
+            -(outputWidth + inputHeight) * 0.5
+          )
+
+          break
+        case 180:
+          context.rotate(Math.PI)
+          context.translate(
+            -(outputWidth + inputWidth) * 0.5,
+            -(outputHeight + inputHeight) * 0.5
+          )
+
+          break
+        case 270:
+          context.rotate((270 * Math.PI) / 180)
+          context.translate(
+            -(outputHeight + inputWidth) * 0.5,
+            (outputWidth - inputHeight) * 0.5
+          )
+      }
+
+      context.drawImage(image, 0, 0)
+      resolve(canvas.toDataURL(imageMimeType, imageQuality))
+    }
+  })
+}
+
+export function base64ToBlob(base64, imageMimeType = 'image/jpeg') {
+  var byteString = atob(base64.split(',')[1]) // eslint-disable-line
+  var ab = new ArrayBuffer(byteString.length)
+  var ia = new Uint8Array(ab)
+
+  for (var i = 0; i < byteString.length; i++) {
+    ia[i] = byteString.charCodeAt(i)
+  }
+  return new Promise((resolve, reject) => resolve({blob: new Blob([ia], {type: imageMimeType}), base64})) // eslint-disable-line
+}
+
+function getExifOrientation(file) {
+  var reader = new FileReader() // eslint-disable-line
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function(e) {
+      var view = new DataView(e.target.result)
+
+      if (view.getUint16(0, false) !== 0xffd8) {
+        resolve(-2)
+      }
+      var length = view.byteLength
+      var offset = 2
+      while (offset < length) {
+        if (view.getUint16(offset + 2, false) <= 8) resolve(-1)
+        var marker = view.getUint16(offset, false)
+        offset += 2
+        if (marker === 0xffe1) {
+          if (view.getUint32((offset += 2), false) !== 0x45786966) {
+            resolve(-1)
+          }
+
+          var little = view.getUint16((offset += 6), false) === 0x4949
+          offset += view.getUint32(offset + 4, little)
+          var tags = view.getUint16(offset, little)
+          offset += 2
+          for (var i = 0; i < tags; i++) {
+            if (view.getUint16(offset + i * 12, little) === 0x0112) {
+              resolve(view.getUint16(offset + i * 12 + 8, little))
+            }
+          }
+        } else if ((marker && 0xff00) !== 0xff00) {
+          break
+        } else {
+          offset += view.getUint16(offset, false)
+        }
+      }
+      resolve(-1)
+    }
+    reader.readAsArrayBuffer(file)
+  })
+}
+
+// function getOrientation(file) {
```

Comment:
is this not being removed for some reason?

Reply:
I'm only saving it commented just in case I break something ! 😝 Deleted! Thanks!